axelor-human-resource/src/main/java/com/axelor/apps/hr/service/batch/BatchReminderTimesheet.java

```diff
@@ -153,11 +153,12 @@ public void generateEmail() {
 
   public void generateAllEmailTemplate() {
 
+    Company company = batch.getMailBatch().getCompany();
     Template template = batch.getMailBatch().getTemplate();
     List<Employee> employeeList = null;
-    if (Beans.get(CompanyRepository.class).all().fetch().size() > 1) {
+    if (company != null) {
       employeeList =
-          Beans.get(EmployeeRepository.class).all().filter("self.timesheetReminder = true").fetch();
+          Beans.get(EmployeeRepository.class).all().filter("self.user.activeCompany.id = ?1 AND self.timesheetReminder = true", company.getId()).fetch();
```

Comment:
here company should be employee.mainEmploymentContract.payCompany.

Indeed, a user can work on different companies.